modules/dxp/apps/portal-reports-engine-console/portal-reports-engine-console-web/src/main/resources/META-INF/resources/admin/definition/edit_definition.jsp

```diff
@@ -208,15 +207,18 @@ else {
 	</aui:button-row>
 </aui:form>
 
-<aui:script>
-	AUI().ready((A) => {
-		Liferay.Report.initialize({
-			namespace: '<portlet:namespace />',
-			parameters:
-				'<%= HtmlUtil.escapeJS(BeanParamUtil.getString(definition, request, "reportParameters")) %>',
-		});
-	});
+<liferay-frontend:component
+	context='<%=
+		HashMapBuilder.<String, Object>put(
+			"namespace", liferayPortletResponse.getNamespace()
+		).put(
+			"parameters", BeanParamUtil.getString(definition, request, "reportParameters")
```

Comment:
Hi @sandrodw3 why are you not escaping here?

I mean, is there a JS escaping happening under the hood when `liferay-frontend:component` is used?

if so, please share with me, otherwise I believe we should keep escaping to avoid XSS (This is my concern [see](https://github.com/liferay-workflow/liferay-portal/pull/1018/files#diff-d08cdcb384d0972f3d08c06d6bfa0af592f003875ab438789850fde18b130111R45))

[this place](https://github.com/liferay/liferay-portal/blob/3eb5e33b485ee4a5a8ecf1d6ab527f9764a15464/modules/apps/asset/asset-publisher-web/src/main/resources/META-INF/resources/configuration/source.jsp#L293) is escaping this way I'm not sure if we should abdicate.

Thank you!

Reply:
Hey @MarinhoFeliphe, I didn't do this part of the code, but I'll take a look!

Reply:
I think you are right, I scaped that param 😄